### PR TITLE
[DUOS-1401][risk=low] Updated google authentication

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadworkbench.atlassian.net/browse/DUOS-1401

This PR replaces the existing google authentication scheme with a purely access-token-based approach.
See https://developers.google.com/identity/oauth2/web/guides/migration-to-gis#gis-only for background.

----
Have you read [Terra's Contributing Guide](https://github.com/DataBiosphere/terra-ui/wiki/Contributor-Guide) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Comment by otchet-broad** [comment]
@rushtong Any idea if other Terra applications are keeping the same button or moving to One Tap?

**Comment by rushtong** [comment]
> @rushtong Any idea if other Terra applications are keeping the same button or moving to One Tap?

Terra has moved to a different oauth library for Azure/B2C support, but not One Tap:
* https://github.com/authts/react-oidc-context
* https://github.com/DataBiosphere/terra-ui/pull/3117

That approach [requires some infrastructure](https://github.com/DataBiosphere/terra-ui/blob/dev/src/libs/auth.js#L25) that we don't have in Consent at the moment so I did not pursue it ... yet. It definitely is something we need to look into, but I see that as a moderately larger task that what I'm trying to do in this PR.

**Review by connorlbark — APPROVED** [review]
As far as I can tell (not an expert on Google sign in), this looks good and works great. 

Pretty minor, but I was wondering about the signin button beneath Data Access Request Application:

image

Originally, this button was the same as if you were logged in. Since it was changed to being a sign in button, I could see this being a little bit confusing/odd from an end user perspective: would it be possible to change it back, or to add some clarifying text, like `Please sign in to submit a Data Access Request` or something.

**Comment by otchet-broad** [comment]
Nice work.

The code led me down a bit of a reading trail and I ended up on this Google page here: 
[https://developers.google.com/identity/oauth2/web/guides/overview](https://developers.google.com/identity/oauth2/web/guides/overview)

Had you seen this and already considered the note on that page?
 
Screen Shot 2022-11-16 at 1 16 13 PM

If I understood it correctly, it suggests an alternate API because of the limited scope being used to Authenticate.

File: public/index.html [hunk 1]
```diff
@@ -22,6 +22,7 @@
   <link rel="icon" type="image/png" href="%PUBLIC_URL%/images/favicon/favicon-32x32.png" sizes="32x32" />
   <link rel="icon" type="image/png" href="%PUBLIC_URL%/images/favicon/favicon-16x16.png" sizes="16x16" />
   <link rel="mask-icon" href="%PUBLIC_URL%/images/favicon/safari-pinned-tab.svg">
+  <script src="https://accounts.google.com/gsi/client" async defer id="gsiClient"></script>
```
Comment (otchet-broad) [inline public/index.html:25]: defer for the win!  well done.

**Comment by rushtong** [comment]
> Originally, this button was the same as if you were logged in. Since it was changed to being a sign in button, I could see this being a little bit confusing/odd from an end user perspective: would it be possible to change it back, or to add some clarifying text, like `Please sign in to submit a Data Access Request` or something.

Great catch, I totally missed that. Will fix it - it should be a button that looks like this:
Screen Shot 2022-11-16 at 1 50 12 PM

**Comment by rushtong** [comment]
> If I understood it correctly, it suggests an alternate API because of the limited scope being used to Authenticate.

I have looked at many of the different use cases. The "Sign In With Google" use case in that link is not exactly what we want. We could make use of it, but then we would only have access to a JWT with basic user information (good). Our use case is to request an access token so we can make back end API calls. If we did use the "Sign In" approach, we would then be required to prompt the user again for an access token since "Sign In" does not grant that. 

As per this content on [their page](https://developers.google.com/identity/oauth2/web/guides/overview):
> Authentication for user sign-in, and authorization to obtain an access token to call Google APIs, now have two separate and distinct user flows; one for [sign-in](https://developers.google.com/identity/gsi/web/guides/overview#how_it_works) and another for [consent](https://developers.google.com/identity/oauth2/web/guides/how-user-authz-works#user_consent) during authorization, with separate user flows to clearly differentiate who you are, from what an app can do.

We really only care about the `consent` case.

**Comment by rushtong** [comment]
@connorlbark - fixed the customized sign in button on the home page under DAR Application, PTAL, thank you!

**Review by otchet-broad — COMMENTED** [review]
Nice work.  This takes care of the immediate need.  Greg and I are continuing to discuss Authentication outside of this PR.

**Review by connorlbark — APPROVED** [review]
New sign in styling looks good to me.

**Review by otchet-broad — APPROVED** [review]

**Pull request merged**